packages/design-systems/src/components/composite/Datagrid/SearchFilter/useCustomFilter.ts

```diff
@@ -324,18 +324,16 @@ export const useCustomFilter = <TData>({
         }
       });
 
-      if (!systemFilter && Object.keys(newGraphQLQueryFilter).length === 0) {
+      if (isNotEmpty(systemFilter) || isNotEmpty(newGraphQLQueryFilter)) {
+        return {
+          and: {
+            ...systemFilter,
+            ...newGraphQLQueryFilter,
+          },
+        };
+      } else {
         return undefined;
       }
```

Comment:
細かいのですが、notのandはちょっと脳がすぐには追いつかないので、お願いしたいです 🙏 

```suggestion
      if (isNotEmpty(systemFilter) && isNotEmpty(newGraphQLQueryFilter)) {
        return undefined
      }
      
      return {
        and: {
          ...systemFilter,
          ...newGraphQLQueryFilter,
        },
      };
```